File: src/python/bitwise-AND-of-numbers-range.py
# 201. Bitwise AND of Numbers Range
# https://leetcode.com/problems/bitwise-and-of-numbers-range/

class Solution:
    def rangeBitwiseAnd(self, left: int, right: int) -> int:
        zero_bits = 0
        while left != right:
            left >>= 1
            right >>= 1
            zero_bits += 1
        return left << zero_bits

File: src/cpp/bitwise-and-of-numbers-range.cpp
//Question: https://leetcode.com/problems/bitwise-and-of-numbers-range/

class Solution {
public:
    //using bitmanipulation
    int rangeBitwiseAnd(int left, int right) {
        int ans = 0;
        while(left<right){
            left >>= 1;
            right >>= 1;
            ans++;
        }
        
        return right<<ans;
    }
};